presto-hive/src/main/java/io/prestosql/plugin/hive/util/AsyncQueue.java

```diff
@@ -45,7 +45,7 @@
     private SettableFuture<?> notFullSignal = SettableFuture.create();
     // This future is completed when the queue transitions from empty to not. But it will be replaced by a new instance of future immediately.
     @GuardedBy("this")
-    private SettableFuture<?> notEmptySignal = SettableFuture.create();
+    protected SettableFuture<?> notEmptySignal = SettableFuture.create();
```

Comment:
Add a protected getter for this that returns `ListenableFuture`. We wouldn't want subclasses to modify the field or to set the future.